File: Baekjoon/S5/C#/S5_집합.cs
https://www.acmicpc.net/problem/11723

using System;
using System.Collections.Generic;
using System.Text;

namespace Baekjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            List<string> S = new List<string>();
            StringBuilder sb = new StringBuilder();

            for (int i = 0; i < n; i++)
            {
                string[] str = Console.ReadLine().Split(' ');
                switch (str[0])
                {
                    case "add":
                        if (!S.Contains(str[1]))
                            S.Add(str[1]);
                        break;

                    case "remove":
                        if (S.Contains(str[1]))
                            S.Remove(str[1]);
                        break;

                    case "check":
                        sb.Append((S.Contains(str[1])) ? "1\n" : "0\n");
                        break;

                    case "toggle":
                        if (S.Contains(str[1]))
                            S.Remove(str[1]);
                        else
                            S.Add(str[1]);
                        break;

                    case "all":
                        if (S.Count > 0)
                            S.RemoveRange(0, S.Count - 1);
                        for (int j = 1; j <= 20; j++)
                            S.Add(j.ToString());
                        break;

                    case "empty":
                        if (S.Count > 0)
                            S.RemoveRange(0, S.Count - 1);
                        break;
                }
            }
            Console.Write(sb);
        }
    }
}

File: Baekjoon/S5/JAVA/S5_집합.java
//https://www.acmicpc.net/problem/11723

import java.io.*;
import java.util.*;

public class Main{
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        int n = Integer.parseInt(br.readLine());
        int m = 0;
        boolean[] s = new boolean[21];
        
        while(n-- > 0){
            String[] str = br.readLine().split(" ");
            
            if(str.length > 1)
                m = Integer.parseInt(str[1]);
                
            switch(str[0]){
                case "add":
                    s[m] = true;
                    break;
                    
                case "remove":
                    s[m] = false;
                    break;
                    
                case "check":
                    sb.append(s[m]? "1\n":"0\n");
                    break;
                    
                case "toggle":
                    s[m] = !s[m];
                    break;
                    
                case "all":
                    for(int i = 1; i <= 20; i++)
                        s[i] = true;
                    break;
                    
                case "empty":
                    for(int i = 1; i <= 20; i++)
                        s[i] = false;
                    break;
            }
        }
        
        System.out.print(sb);
    }
}